src/include/duckdb/function/macro_function.hpp

```diff
@@ -32,6 +37,11 @@ class MacroFunction {
 	//! The default parameters and their associated values
 	unordered_map<string, unique_ptr<ParsedExpression>> default_parameters;
 
+	// if true then we have a query_node and not a regular expression
+	// bool isQuery;
+	//! The main query node
+	unique_ptr<QueryNode> query_node;
```

Comment:
Rather than expand the `MacroFunction`, I think we should add a new `TableMacroFunction` class, for a few reasons:

* Name conflicts should be on the *table-function* level, rather than on the *scalar function* level (i.e. the `TableMacroFunction` objects need to be stored in the `table_functions` CatalogSet, not in the `functions` catalog set).
* The separate `BindToMacro` function should not be required, and we can bind macros (mostly) the same as regular table functions (with some additional checks/code).
* This is generally cleaner, since it means we don't need to check if we are using the correct type of macro function in each context.

Perhaps there can be a `BaseMacroFunction` that both the regular `MacroFunction` and the `TableMacroFunction` inherit from, to avoid having to duplicate a lot of functionality.

Reply:
Yep that makes sense